windows-agent/internal/config/config_test.go

```diff
@@ -638,6 +639,83 @@ func loadChecksums(t *testing.T, confDir string) (string, string) {
 	return fileData.Subscription.Checksum, fileData.Landscape.Checksum
 }
 
+func TestNotify(t *testing.T) {
+	ctx := context.Background()
+	if wsl.MockAvailable() {
+		t.Parallel()
+		ctx = wsl.WithMock(ctx, wslmock.New())
+	}
+
+	db, err := database.New(ctx, t.TempDir(), nil)
+	require.NoError(t, err, "Setup: could not create empty database")
+
+	_, dir := setUpMockSettings(t, ctx, db, untouched, false)
+	c := config.New(ctx, dir)
+
+	var notifyCount atomic.Int32
+	var wantNotifyCount int32
+
+	c.Notify(func() { notifyCount.Add(1) })
+
+	err = c.SetUserSubscription(ctx, "TOKEN_1")
+	require.NoError(t, err, "SetUserSubscription should return no error")
+	wantNotifyCount++
+
+	eventually(t, notifyCount.Load, func(got int32) bool { return wantNotifyCount == got },
+		time.Second, 100*time.Millisecond, "Attached function should have been called after changing the pro token")
+
+	err = c.SetLandscapeAgentUID(ctx, "UID_1")
+	require.NoError(t, err, "SetLandscapeAgentUID should return no error")
+	wantNotifyCount++
+
+	eventually(t, notifyCount.Load, func(got int32) bool { return wantNotifyCount == got },
+		time.Second, 100*time.Millisecond, "Attached function should have been called after changing the landscape UID")
+
+	err = c.UpdateRegistryData(ctx, config.RegistryData{UbuntuProToken: "TOKEN_2"}, db)
+	require.NoError(t, err, "UpdateRegistryData should return no error")
+	wantNotifyCount++
+
+	eventually(t, notifyCount.Load, func(got int32) bool { return wantNotifyCount == got },
+		time.Second, 100*time.Millisecond, "Attached function should have been called after changing registry data")
+}
+
+// eventually solves the main issue with 'require.Eventually': you don't know what failed.
+// See what happens if you try to build the 'want vs. got' error message:
+//
+//	  require.Eventuallyf(t, func()bool{ return 5==got() },
+//			t, trate, "Mismatch: wanted %d but got %d", 5, got())
+//	                                                       ^^^
+//														   OUTDATED!
+//
+// Got is passed by value so we have the value obtained at t=0, not at t=timeout.
+//
```

Comment:
Should we propose a patch to the upstream require package?

Reply:
My solution is less generic as it requires an intermediate object:
```
predicate(getter())
```

Upstream does not require it:
```
predicate()
```
And this object must be representable as a string. In our specific case, this was not a problem. But it's a bit too restrictive for upstream I believe.